mkfs/mkfs.c

```diff
@@ -636,6 +638,22 @@ int main(int argc, char *argv[])
 			if (*optarg != '\0' && *optarg != '\r')
 				ui.guid = optarg;
 			break;
+		case 's':
+			ret = parse_size(optarg);
+			if (ret < 0)
+				goto out;
+			else if (ret & (ret - 1)) {
+				exfat_err("sector size(%d) is not a power of 2\n",
+					ret);
+				goto out;
+			} else if ((ret & 0x1e00) == 0) {
```

Comment:
clusters being smaller than sectors is already checked here:

https://github.com/exfatprogs/exfatprogs/blob/0bf46acbd26483f728231d54416a23637fce764c/mkfs/mkfs.c#L460-L464

the only use of `ui->cluster_size` before this is in `exfat_get_blk_dev_info` where it's used to populate the `num_clusters` field. i plan on removing this field in a later PR because the number of clusters is not a property of the underlying volume but of the exfat filesystem

Reply:
Okay, 

or validating 

> clusters being smaller than sectors is already checked here:
> 
> https://github.com/exfatprogs/exfatprogs/blob/0bf46acbd26483f728231d54416a23637fce764c/mkfs/mkfs.c#L460-L464
> 
> the only use of `ui->cluster_size` before this is in `exfat_get_blk_dev_info` where it's used to populate the `num_clusters` field. i plan on removing this field in a later PR because the number of clusters is not a property of the underlying volume but of the exfat filesystem

Okay, Or it also looks good to compare the cluster size with the sector size right after parsing parameters.